src/lib/cost_estimation/cost_estimator_logical.cpp

```diff
@@ -14,66 +16,106 @@
 #include "types.hpp"
 #include "utils/assert.hpp"
 
+namespace {
+
+using namespace hyrise;  // NOLINT(build/namespaces)
+
+float expression_cost_multiplier(const std::shared_ptr<AbstractExpression>& expression) {
+  auto multiplier = 0.0f;
+
+  // Number of different columns accessed to factor in expression complexity. Also add a factor for correlated
+  // subqueries since we have to evaluate the subquery for each tuple again. In the past, we added to the factor for
+  // each expression in the predicate. This led to too pessimistic cost estimations for PredicateNodes compared to
```

Comment:
People reading that in a year will have a hard time to understand what has been done and why things have been too pessimistic? Rather just write what you do now and why?